manifests/repo.pp

```diff
@@ -32,15 +32,14 @@
 
   case $::osfamily {
     'Debian': {
-      if !defined(Class['apt']) {
-        class { 'apt': }
-      }
+      include ::apt
+      Class['apt::update'] -> Package<| tag == 'elasticsearch-package' |>
```

Comment:
If I'm understanding things correctly, we can't be certain that resources in `apt::update` (in this case the `exec` resource) will be handled before package installs since `apt::update` is included, but not [contained](https://docs.puppetlabs.com/puppet/latest/reference/lang_containment.html), inside of  the `apt` class (containing it there would likely cause other issue). Adding a notify would ensure that `apt::update` is handled after `apt::source`, but wouldn't ensure it all happens prior to the handling the `package` resource.

There is however a cleaner method than a tag that I'll push up shortly.